test/hotspot/jtreg/gc/parallel/TestAlwaysPreTouchBehavior.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2014, 2024, Alibaba Group Holding Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package gc.parallel;
+
+/**
+ * @test TestAlwaysPreTouchBehavior
+ * @summary Tests AlwaysPreTouch Bahavior, pages of java heap should be pretouched with AlwaysPreTouch enabled. This test reads RSS of test process, which should be bigger than heap size(1g) with AlwaysPreTouch enabled.
+ * @requires vm.gc.Parallel
+ * @requires os.family == "linux" & os.maxMemory > 2G
+ * @library /test/lib
```

Comment:
The test only works in release builds as the debug builds incidentally always pretouch bases.
I.e. add `@requires vm.debug != true`.
There does not seem to be a `@requires vm.release` or something.

Reply:
Another option is to `@require vm.debug` and add `-XX:-ZapUnusedHeapArea` to the run options ( suggested by @albertnetymk ).